src/Compilers/CSharp/Test/Emit2/Semantics/ParamsCollectionTests.cs

```diff
@@ -778,5 +779,1104 @@ public static void M2(params ushort[] x)
+
+        // Ambiguous for inline collection expression, but 'int' is a better conversion target than 'object' in params case
+        [InlineData("System.Span<T>", "System.ReadOnlySpan<object>", "System.Span<System.Int32>")] // cannot convert from object to int
+
+        [InlineData("RefStructCollection<T>", "T[]", null, new[] { CollectionExpressionTests.example_RefStructCollection })] // rule requires span
+
+        // Ambiguous for inline collection expression, but 'int' is a better conversion target than 'object' in params case
+        [InlineData("RefStructCollection<T>", "RefStructCollection<object>", "RefStructCollection<System.Int32>", new[] { CollectionExpressionTests.example_RefStructCollection })] // rule requires span
+        // Ambiguous for inline collection expression, but 'int' is a better conversion target than 'object' in params case
+        [InlineData("RefStructCollection<int>", "GenericClassCollection<object>", "RefStructCollection<System.Int32>", new[] { CollectionExpressionTests.example_RefStructCollection, CollectionExpressionTests.example_GenericClassCollection })] // rule requires span
+        // Ambiguous for inline collection expression, but 'int' is a better conversion target than 'object' in params case
+        [InlineData("RefStructCollection<object>", "GenericClassCollection<int>", "GenericClassCollection<System.Int32>", new[] { CollectionExpressionTests.example_RefStructCollection, CollectionExpressionTests.example_GenericClassCollection })] // cannot convert object to int
+        // Ambiguous for inline collection expression, but 'int' is a better conversion target than 'object' in params case
+        [InlineData("RefStructCollection<int>", "NonGenericClassCollection", "RefStructCollection<System.Int32>", new[] { CollectionExpressionTests.example_RefStructCollection, CollectionExpressionTests.example_NonGenericClassCollection })] // rule requires span
+
+        [InlineData("GenericClassCollection<T>", "T[]", null, new[] { CollectionExpressionTests.example_GenericClassCollection })] // rule requires span
+        [InlineData("NonGenericClassCollection", "object[]", null, new[] { CollectionExpressionTests.example_NonGenericClassCollection })] // rule requires span
+        [InlineData("System.ReadOnlySpan<T>", "object[]", "System.ReadOnlySpan<System.Int32>")]
+        [InlineData("System.ReadOnlySpan<T>", "long[]", "System.ReadOnlySpan<System.Int32>")]
+
+        // Ambiguous for inline collection expression, but 'int' is a better conversion target than 'short' in params case (exact target)
+        [InlineData("System.ReadOnlySpan<T>", "short[]", "System.ReadOnlySpan<System.Int32>")] // cannot convert int to short
+        // Ambiguous for inline collection expression, but 'int' is a better conversion target than 'long' in params case
+        [InlineData("System.ReadOnlySpan<long>", "T[]", "System.Int32[]")] // cannot convert long to int
```

Comment:
This one is particularly surprising. Is part of the open question you have for LDM?

Reply:
> This one is particularly surprising. Is part of the open question you have for LDM?

I do not find this surprising for this scenario at all. The better conversion rule can, of course, have a different effect and that is explicitly called out in the specification. Directly in the relevant section, with examples, rather than as an open question.